File: 08/simple_locate.rb
require 'selenium-webdriver'

dr = Selenium::WebDriver.for :chrome
file_path = 'file:///' + File.expand_path(File.join('.', 'form.html'))
puts file_path
dr.get file_path

# by id
dr.find_element(:id, 'inputEmail').click

# by name
dr.find_element(:name, 'password').click

# by tagname
puts dr.find_element(:tag_name, 'form')[:class]

# by class_name
e = dr.find_element(:class, 'controls')
dr.execute_script('$(arguments[0]).fadeOut().fadeIn()', e)
sleep 1

# by link text
link = dr.find_element(:link_text, 'register')
dr.execute_script('$(arguments[0]).fadeOut().fadeIn()', link)
sleep 1

# by partial link text
link = dr.find_element(:partial_link_text, 'reg')
dr.execute_script('$(arguments[0]).fadeOut().fadeIn()', link)
sleep 1

# by css selector
div = dr.find_element(:css, '.controls')
dr.execute_script('$(arguments[0]).fadeOut().fadeIn()', div)
sleep 1

# by xpath
dr.find_element(:xpath, '/html/body/form/div[3]/div/label/input').click

sleep 2
dr.quit

File: 08/SimpleLocate.java
import java.io.File;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.JavascriptExecutor;


public class SimpleLocate {

	public static void main(String[] args) throws InterruptedException {
		WebDriver dr = new ChromeDriver();
		
		File file = new File("src/form.html");
		String filePath = "file:///" + file.getAbsolutePath();
		System.out.printf("now accesss %s \n", filePath);
		
		dr.get(filePath);
		Thread.sleep(2000);
		
//		by id
		dr.findElement(By.id("inputEmail")).click();
		Thread.sleep(1000);
		
//		by name
		dr.findElement(By.name("password"));
		Thread.sleep(1000);
		
//		by tagname
		String classOfForm = dr.findElement(By.tagName("form")).getAttribute("class");
		System.out.printf("%s\n", classOfForm);
		Thread.sleep(1000);
		
//		by link text
		WebElement link = dr.findElement(By.linkText("register"));
		((JavascriptExecutor)dr).executeScript("$(arguments[0]).fadeOut().fadeIn()", link);
		Thread.sleep(1000);

//		by partial link test
		WebElement sameLink = dr.findElement(By.partialLinkText("reg"));
		((JavascriptExecutor)dr).executeScript("$(arguments[0]).fadeOut().fadeIn()", sameLink);
		Thread.sleep(1000);
		
//		by css selector
		WebElement div = dr.findElement(By.cssSelector(".controls"));
		((JavascriptExecutor)dr).executeScript("$(arguments[0]).fadeOut().fadeIn()", div);
		Thread.sleep(1000);
		
//		by xpath
		dr.findElement(By.xpath("/html/body/form/div[3]/div/label/input")).click();
		Thread.sleep(1000);
		
		System.out.println("browser will be close");
		dr.quit();	
	}

}
